torch/_inductor/codegen/cpp.py

```diff
@@ -1231,6 +1231,17 @@ def bitwise_left_shift(a, b):
     def bitwise_right_shift(a, b):
         return f"{a} >> {b}"
 
+    @staticmethod
+    def remainder(a, b):
+        assert (
+            a.dtype == b.dtype
+        ), "remainder vec implementation expect the same inputs' dtype."
+        if is_float_dtype(a.dtype):
+            return f"{a} - ({a} / {b}).floor() * {b}"
```

Comment:
why using these two lines? I believe this is not correct for very large numbers due to rounding issues.
At any rate, if we want to have a shortcut of this kind anywhere, I think it should go in `floordiv`, but again, I don't think this is correct.

Reply:
> why using these two lines? I believe this is not correct for very large numbers due to rounding issues.

Hi @lezcano, thanks for the comment. I am not clearly understanding the rounding issues you mentioned here. could you kindly give a example to debug this issue?

> At any rate, if we want to have a shortcut of this kind anywhere, I think it should go in `floordiv`, but again, I don't think this is correct.

The implementation of `floordiv` only counts of input with `integer type`. If user compile `torch.div(a, b rounding_mode="floor")` with `float` data type, it actually lowering as `truediv` and `floor`.
```
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0] scheduling ComputedBuffer(name='buf0', layout=FixedLayout('cpu', torch.float32, size=[128], stride=[1]), data=Pointwise(
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]   'cpu',
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]   torch.float32,
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]   def inner_fn(index):
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]       i0 = index
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]       tmp0 = ops.load(arg1_1, i0)
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]       tmp1 = ops.load(arg0_1, i0)
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]       tmp2 = tmp0 / tmp1
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]       tmp3 = ops.floor(tmp2)
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]       return tmp3
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]   ,
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]   ranges=[128],
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]   origin_node=div,
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0]   origins={div}
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1681] [0/0] ))
V0710 03:01:36.218000 139631037911424 torch/_inductor/scheduler.py:1757] [0/0] scheduling output buf0

```

Reply:
I thought that if the output was large enough there could be issues where div+floor would not give the right result, but after a second thought I think it might be alright

Now, why are we reinventing the wheel here rather than:
- Dispatching to the eager implementation: https://github.com/pytorch/pytorch/blob/cb2bce98de5bba1c5415a08d7b17c608f6c77b62/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp#L246-L267
- Remove this float case and always always dispatch this to `return f"{a} - ({CppVecOverrides.floordiv(a, b)}) * {b}"` (or floordiv in terms of fmod, like in eager)